pkg/cmd/roachtest/roachtestutil/mixedversion/mutators.go

```diff
@@ -383,3 +383,57 @@ func (m clusterSettingMutator) changeSteps(
 
 	return steps
 }
+
+const (
+	// PanicNode is a mutator that will randomly cause a node to crash
+	// during the test, before restarting the node within the same step.
+	PanicNode = "panic_node"
+)
+
+type panicNodeMutator struct {
```

Comment:
Something to consider but not necessarily implement in this PR, is how you can make adding future failure mutators easier.

For example, if we wanted to add a failure that crashed a node by signaling `SIGKILL` instead, it would look essentially identical to `panicNodeMutator` with the exception of it inserting a `signalNodeStep` instead.

The rules for inserting failure mutators should be the same I think, e.g. we don't want to do concurrently with restart steps, if we alter node liveness we need to update some state, etc. Mimicking how we have cluster mutators that share common helpers could be the way to go, but like I mentioned not something that needs to be done yet.